src/cpp/session/modules/SessionCodeTools.R

```diff
@@ -1488,10 +1488,41 @@
  
 })
 
+# If a function internally calls out to one of these functions,
+# we assume that it may perform non-standard evaluation, and avoid
+# attempting certain diagnostics on its usages.
+#
+# Keep in sync with makeNsePrimitives() in RSexp.cpp
 .rs.setVar("nse.primitives", c(
-   "quote", "substitute", "match.call", "eval.parent",
-   "enquote", "bquote", "evalq", "lazy_dots", "compat_as_lazy_dots",
-   "select_vars", "quo", "quos", "enquo", "named_quos"
+   
+   # base R primitives
+   "bquote",
+   "enquote",
+   "eval.parent",
+   "evalq",
+   "expression",
+   "library",
+   "match.call",
+   "quote",
+   "require",
+   "subset",
+   "substitute",
+   "sys.call",
+   "sys.calls",
+   "sys.frame",
+   "sys.frames",
+   "sys.function",
+   "sys.parent",
+   
+   # rlang tidy evaluation
+   "enexpr",
+   "enquo",
```

Comment:
I think `tidyr::pivot_longer()` will still be flagged with "symbol not in scope"

```r
library(tidyr)

df <- tibble(x = c("a", "b", "a"))

pivot_longer(df, x)
```

If you look at `tidyr:::pivot_longer.data.frame()`, the `cols` argument is passed along using `{{ cols }}` which is semantically the same as `!!enquo(cols)` but `enquo()` doesn't appear.

Reply:
Perhaps we could also detect usages of `{{}}`? I'm guessing those are going to be relatively uncommon outside of tidy evaluation.